csharp/src/Apache.Arrow/Memory/ILargeOwnableAllocation.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Apache.Arrow.Memory
+{
+    internal interface ILargeOwnableAllocation
```

Comment:
I don't think this class is needed yet, let's remove it and we can always add it back later if needed when looking at C Data Interface support.